File: backend/src/main/java/com/okaya/service/ClientService.java
package com.okaya.service;

import com.okaya.model.Client;
import com.okaya.repository.ClientRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

/**
 * Service pour la gestion des clients
 */
@Service
public class ClientService {
    
    @Autowired
    private ClientRepository clientRepository;
    
    /**
     * Récupérer tous les clients
     */
    public List<Client> getAllClients() {
        return clientRepository.findAll();
    }
    
    /**
     * Récupérer un client par son ID
     */
    public Optional<Client> getClientById(Long id) {
        return clientRepository.findById(id);
    }
    
    /**
     * Récupérer un client par son code
     */
    public Optional<Client> getClientByCode(String codeClient) {
        return clientRepository.findByCodeClient(codeClient);
    }
    
    /**
     * Créer ou mettre à jour un client
     */
    public Client saveClient(Client client) {
        return clientRepository.save(client);
    }
    
    /**
     * Supprimer un client
     */
    public void deleteClient(Long id) {
        clientRepository.deleteById(id);
    }
}

File: frontend/src/app/services/client.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Client } from '../models/client.model';

/**
 * pour la gestion des clients
 */
@Injectable({
  providedIn: 'root'
})
export class ClientService {
  private apiUrl = 'http://localhost:8080/api/clients';

  constructor(private http: HttpClient) {}

  /**
   * Récupérer tous les clients
   */
  getAll(): Observable<Client[]> {
    return this.http.get<Client[]>(this.apiUrl);
  }

  /**
   * Récupérer un client par ID
   */
  getById(id: number): Observable<Client> {
    return this.http.get<Client>(`${this.apiUrl}/${id}`);
  }

  /**
   * Créer un nouveau client
   */
  create(client: Client): Observable<Client> {
    return this.http.post<Client>(this.apiUrl, client);
  }

  /**
   * Mettre à jour un client
   */
  update(id: number, client: Client): Observable<Client> {
    return this.http.put<Client>(`${this.apiUrl}/${id}`, client);
  }

  /**
   * Supprimer un client
   */
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
